test/mocks/data/firestore_challenge.dart

```diff
@@ -0,0 +1,49 @@
+import "package:cloud_firestore/cloud_firestore.dart";
+import "package:proxima/models/database/challenge/challenge_firestore.dart";
+import "package:proxima/models/database/post/post_id_firestore.dart";
+import "package:proxima/models/database/user/user_firestore.dart";
+import "package:proxima/models/database/user/user_id_firestore.dart";
+
+import "challenge_data.dart";
+
+class FirestoreChallengeGenerator {
+  int _id = 0;
+
+  ChallengeFirestore generateChallenge(bool done, Duration extra) {
+    final data = ChallengeGenerator.generate(done, extra);
+    final id = PostIdFirestore(value: (_id++).toString());
```

Comment:
I was thinking that we could use `package:uuid/uuid.dart` for a robust uuid generation, though it is not causing any issue for now.
(but might be nice if we go big :) )

Reply:
I don't agree, this generator is meant for small scale testing. While debugging tests and looking at the current state of documents in the database, I find it much easier to have simple small ids than long random strings.

If we find that we need to use robust uuids in the future we could add them, or just call the methods in the repositories. I don't think we should change anything now though as it would make debugging harder.